src/vos/vos_obj.c

```diff
@@ -744,6 +747,27 @@ key_iter_fetch(struct vos_obj_iter *oiter, vos_iter_entry_t *ent,
 	if (rc != 0)
 		return rc;
 
+	if (check_existence && oiter->it_iter.it_filter_cb != NULL) {
+		desc.id_type = oiter->it_iter.it_type;
+		desc.id_key = ent->ie_key;
+		acts = 0;
+		start_seq = vos_sched_seq();
+		rc = oiter->it_iter.it_filter_cb(vos_iter2hdl(&oiter->it_iter), &desc,
+						 oiter->it_iter.it_filter_arg,
+						 &acts);
+		if (rc != 0)
+			return rc;
+		if (acts & VOS_ITER_CB_ABORT)
+			return VOS_ITER_CB_ABORT;
+		if (start_seq != vos_sched_seq())
+			return VOS_ITER_CB_YIELD;
+		if (acts != 0) {
+			if (acts & VOS_ITER_CB_SKIP)
+				return IT_OPC_NEXT;
+			D_ASSERTF(0, "Invalid acts returned from iterator filter %x\n", acts);
+		}
```

Comment:
because you have two code blocks which do the same check, can we have an inline function to do these checks?